### ISSUE TITLE
Login with github or google redirects to sign up

### ISSUE THREAD

**Issue by rbbeeston** [boostorg/website-v2#1293]
When logging into the site, when selecting github or google to login, it's redirecting to signup.
https://github.com/user-attachments/assets/d66f555b-8c80-4116-90f6-253870408d7e

**Comment by vinniefalco** [comment]
ugh this again

**Comment by sdarwin** [comment]
There is no problem. The message was expected.    Stage is separate from Production.  It has a different DB.     Rob's account on stage is Google, so it complained about switching Social Auth to Github, and asked you to log in to change it.  See screenshot.

![Screenshot from 2024-09-26 14-15-01](https://github.com/user-attachments/assets/943ccdbc-208f-451a-b7dd-2bddb32c035e)

**Comment by rbbeeston** [comment]
ok, so if you attempt to log in with the wrong account type, it switches to sign in. I get it, and that's fine, but it seems like there should be some type of notification, pop-up or something to let the user know.

**Comment by sdarwin** [comment]
From the .mov file you posted _there is a notification_ in red and white text.

If that workflow can be improved, you might even suggest it upstream to https://github.com/pennersr/django-allauth

**Comment by rbbeeston** [comment]
yes, but that's not the part that initially caused the confusion. in the first part of the video, I am on a "log in" screen when I click on the "Log In" the next screen comes up that says "Sign In" at the top. (note I circled it with the mouse) 

This is where I thought there was a problem, because if I think I'm logging in, and then it takes me to a page that says "Sign up" at the top, I feel like I've been sent to the wrong place. but I click the button anyway, and it says in RED:

"An account already exists with that email already exists, please sign in"

And I think to myself, "I know that, That's what I'm ****ing trying to do!"

So based on what I've seen, I was trying to go through the process correctly, and the system tells me I'm doing it wrong. I had no way of knowing I was signing in with the wrong account type, and from what I could see, it was the system not letting me log in because the workflow was poor.

I can certainly pass it along to the devs. Is it ok for me to send the video along as well?

**Comment by sdarwin** [comment]
You might expect that the system provide a more clear explanation such as "rob@cppalliance.org already has an account using Google.  Log in with your Google credentials instead."    The current message is generic, doesn't seem to know about that exact information, and merely says "you already have an account".

There is a concern - more detailed messages start to reveal too much about a user during random login attempts.

**Comment by rbbeeston** [comment]
I don't think it's necessary to specify the account (although we are showing the last login account on the login page ) just making sure that the login screen says "Log In" at the top and on the button instead of "Sign Up"

**Comment by sdarwin** [comment]
"Log in" instead of "Sign up"

1.  You didn't reply to the idea I posted in the other issue:  twitter, github, stackexchange all show **both** buttons: "log in" "sign up".     That would be a standard industry practice, and easy to implement.   It doesn't need to be either/or.
2.   Or, since it sometimes show "last log in", that means the website knows you've logged in before.    If that information can be leveraged, based on existing session state, then at least new cookies don't have to be added.

**Comment by rbbeeston** [comment]
Ok, to go a little bit deeper into those:
I looked at all three of those sites logins, and there were 3 different ways of doing it.
X (twitter) - used a single login/signup button that was most like what we are doing now all on one page, the nice thing is that it simplifies it, and makes the login option more clear, however, that's the default page, as you need a login to access the site.
![Screenshot 2024-09-27 at 8 30 50 AM](https://github.com/user-attachments/assets/66b53c22-278e-4a8d-a5a5-60da97c91c34)
Github - Uses two buttons in the header. I'm ok with this route, but we may want to consider how it would look. perhaps make them more button like, or use smaller text.
StackExchange - This one freaked me out, and I doubt our users would appreciate it. When I clicked on the sign up button, it automatically created an account using my google profile, without checking with me. At first I wondered if I had created an account previously, but it was with my cpp email. I didn't like that I suddenly had an account with my name and email without some verification first.

Overall, I prefer the 'X' layout, and it could be done without major modifications to what we are already doing.

Another option would be to have something different than "Join" or "Login" like "Account" or "Profile"

Regarding #2, do we want to remove the "last login" notification, to improve the security? If not, then I don't see any reason not to just use that information to simplify the login process.

**Comment by sdarwin** [comment]
> X (twitter) - used a single login/signup button

Not a single login/signup button. That screenshot above shows "Create account" and "Sign In".  Two buttons.

> Github - Uses two buttons

Right.

> StackExchange - This one freaked me out,

Two.

**Comment by rbbeeston** [comment]
What's funny about stack exchange, is it's different depending on if you are on stackexchange.com or meta.stackexchange.com
![Screenshot 2024-09-27 at 9 23 58 AM](https://github.com/user-attachments/assets/e4faffec-f2e7-4473-a054-e796ea6ee234)
![image](https://github.com/user-attachments/assets/422bbd5c-f5de-4760-b6bb-c25d4bae7e1c)
